src/components/SearchToolbar/CategoryButton.js

```diff
@@ -1,64 +1,163 @@
+import type { RouterHistory } from 'react-router-dom';
 
-import IconButton from '../IconButton';
-import * as icons from '../icons/mainCategories';
+import CombinedIcon from './CombinedIcon';
 import colors from '../../lib/colors';
-import { t } from 'c-3po';
+import IconButton from '../IconButton';
+import CloseIcon from '../icons/actions/Close';
+import type { YesNoLimitedUnknown, YesNoUnknown } from '../../lib/Feature';
+import { getQueryParams, newLocationWithReplacedQueryParams } from '../../lib/queryParams';
+import { isFiltered } from '../../lib/Feature';
+
 
 type Props = {
   name: string,
   id: string,
   className: string,
   hidden: boolean,
-  onFocus: ((event: UIEvent) => void),
-  onBlur: ((event: UIEvent) => void),
-  onKeyDown: ((event: UIEvent) => void),
+  showCloseButton: boolean,
+  hasCircle?: boolean,
+  accessibilityFilter?: YesNoLimitedUnknown[],
+  toiletFilter?: YesNoUnknown[],
+  onFocus?: ((event: UIEvent) => void),
+  onBlur?: ((event: UIEvent) => void),
+  onKeyDown?: ((event: UIEvent) => void),
+  history: RouterHistory,
 };
 
-
-const evenMoreTransparentLinkColor = hsl(colors.linkBackgroundColorTransparent);
-evenMoreTransparentLinkColor.opacity *= 0.5;
-
 const StyledNavLink = styled(NavLink)`
   border-radius: 5px;
 
+  figure {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  .circle {
+    background-color: ${colors.tonedDownSelectedColor};
+    margin: 2.5px;
+    svg.icon {
+      g, path, circle, rect {
+        fill: white;
+      }
+    }
+  }
+  
+  svg.icon {
+    width: 21px;
+    height: 21px;
+    top: initial;
+    left: initial;
+  }
+
   &.active {
-    background-color: ${evenMoreTransparentLinkColor};
+    background-color: ${colors.evenMoreTransparentLinkColor};
     .circle {
       background-color: ${colors.selectedColor};
     }
   }
+
   &:hover, &:focus {
     background-color: ${colors.linkBackgroundColorTransparent};
+    .circle {
+      background-color: ${colors.halfTonedDownSelectedColor};
+    }
     &.active {
       .circle {
-        background-color: ${colors.darkSelectedColor};
+        background-color: ${colors.tonedDownSelectedColor};
+      }
+    }
+  }
+
+
+  &.is-horizontal {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: space-between;
+    padding: 0 1em;
+    min-height: 3rem;
+    .icon-button {
+      flex: 1;
+      .accessibilities {
+        width: 70px;
+      }
+      .caption {
+        color: ${colors.darkSelectedColor};
+      }
+    }
+  }
+
+  &:not(.is-horizontal) {
+    flex-direction: column;
+    .icon-button {
+      flex-direction: column;
+      .accessibilities {
+        justify-content: center;
+      }
+      .caption {
+        font-size: 0.80em;
+        margin-top: 0.5em;
+        color: ${colors.darkSelectedColor};
       }
     }
   }
 `;
 
 
+function urlForFilters({ history, id, accessibilityFilter, toiletFilter, showCloseButton }) {
```

Comment:
This method exists twice with slightly different implementations, can they be consolidated?